spec/mongoid/indexable/specification_spec.rb

```diff
@@ -154,6 +156,68 @@
                                    })
       end
     end
+
+    context "with aliased field options nested inside $ operators" do
+
+      let(:spec) do
+        described_class.new(
+          Band,
+          { name: 1, title: 1, years: -1 },
+          partial_filter_expression: {
+            '$foo' => { years: { '$lte' => 50 } },
+            '$bar' => [
+              { views: { '$gt' => 100 } },
+              { years: { '$lte' => 50 } }
+            ]
+          }
+        )
+      end
+
+      it "sets the class" do
+        expect(spec.klass).to eq(Band)
+      end
+
+      it "normalizes the key" do
+        expect(spec.key).to eq(name: 1, title: 1, y: -1)
+      end
+
+      it "normalizes the options" do
+        expect(spec.options).to eq(partial_filter_expression: {
+          '$foo': { y: { '$lte' => 50 } },
+          '$bar': [
+            { views: { '$gt' => 100 } },
+            { y: { '$lte' => 50 } }
+          ]
+        })
+      end
+    end
+
+    context "with aliased field options double-nested" do
+
+      let(:spec) do
+        described_class.new(
+          Band,
+          { name: 1, title: 1, years: -1 },
+          partial_filter_expression: {
+            '$foo' => { years: { years: { '$lte' => 50 } } },
+          }
+        )
+      end
+
+      it "sets the class" do
+        expect(spec.klass).to eq(Band)
+      end
+
+      it "normalizes the key" do
+        expect(spec.key).to eq(name: 1, title: 1, y: -1)
+      end
+
+      it "normalizes the options" do
+        expect(spec.options).to eq(partial_filter_expression: {
+          '$foo': { y: { years: { '$lte' => 50 } } },
```

Comment:
Can you explain to me why you've only decided to expand the alias of one level below the operator?

Reply:
Because there is no real-world syntax in MongoDB which allows fields to be used this way.

I'd like to again ask that the focus here be on *real-world behavior*, and not nonsensical hypotheticals. This PR covers every real-world case and more of the nonsense than it should, in my opinion.

Let's merge.